### PR TITLE
DTGB-210: Added single image - image gallery and reorganised atoms and molecules.

### PR DESCRIPTION
**Pull request by delrueba**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the style guide CHANGELOG accordingly.
- [ ] I have updated the gent_base CHANGELOG accordingly.

### CODE REVIEW

File: styleguide/components/21-atoms/buttons/button-primary/button-primary.config.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "status": "ready",
   "handle": "button-primary",
   "default": "button-small",
+  "collated": "true",
   "order": 1,
```
Comment (meirege) [inline styleguide/components/21-atoms/buttons/button-primary/button-primary.config.json:8]: If dark and alert buttons don't get an order, this shouldn't either.

File: styleguide/components/21-atoms/buttons/button-secondary/button-secondary.config.json [hunk 2]
```diff
@@ -3,8 +3,9 @@
   "name": "Button secondary",
   "status": "ready",
   "handle": "button-secondary",
+  "collated": "true",
   "default": "button-small",
-  "order": 1,
+  "order": 2,
```
Comment (meirege) [inline styleguide/components/21-atoms/buttons/button-secondary/button-secondary.config.json:8]: If dark and alert buttons don't get an order, this shouldn't either.

File: styleguide/components/21-atoms/headings/heading-6/_heading-6.scss [hunk 3]
```diff
@@ -1,8 +0,0 @@
-h6,
-.h6,
-%h6 {
-  @include theme('color', 'color-secondary', 'heading-6-color');
-
-  margin-bottom: .5rem;
-  font-weight: 600;
-}
```
Comment (meirege) [inline styleguide/components/21-atoms/headings/heading-6/_heading-6.scss:8]: Maybe add to the readme that h6's aren't being used throughout the style guide?

File: styleguide/components/21-atoms/inputs/inputs.config.json [hunk 4]
```diff
@@ -1,4 +1,4 @@
 {
   "title": "Inputs",
-  "order": 7
+  "hidden": true
```
Comment (meirege) [inline styleguide/components/21-atoms/inputs/inputs.config.json:3]: Why hide the input fields? Are we only going to show them inside their respective molecules?

File: styleguide/components/41-organisms/header/header--with-banner.twig [hunk 5]
```diff
@@ -42,10 +42,9 @@
...
-"src": page_wide_image_src,
-"alt_text": page_wide_image_alt_text,
-"modifier": page_wide_image_modifier,
-"description": page_wide_image_description,
-"h2_text": page_wide_iamge_h2_text
+{% include '@figure' with {
+"figure_image_src": page_wide_image_src,
+"figure_image_alt_text": page_wide_image_alt_text,
+"modifier": 'page-wide-image',
+"figure_figcaption": page_wide_image_description
```
Comment (meirege) [inline styleguide/components/41-organisms/header/header--with-banner.twig]: Indentation.

File: styleguide/components/41-organisms/header/header--with-menu-and-banner.twig [hunk 6]
```diff
@@ -54,11 +54,10 @@
...
-  "src": page_wide_image_src,
-  "alt_text": page_wide_image_alt_text,
-  "modifier": page_wide_image_modifier,
-  "description": page_wide_image_description,
-  "h2_text": page_wide_image_h2_text
+{% include '@figure' with {
+  "figure_image_src": page_wide_image_src,
+  "figure_image_alt_text": page_wide_image_alt_text,
+  "modifier": 'page-wide-image',
+  "figure_figcaption": page_wide_image_description
```
Comment (meirege) [inline styleguide/components/41-organisms/header/header--with-menu-and-banner.twig:61]: Indentation.

File: styleguide/components/71-pages/general/info-page/info-page.twig [hunk 7]
```diff
@@ -36,12 +36,11 @@
...
-  "src": "https://via.placeholder.com/1600x350",
-  "alt_text": "Page wide image",
-  "modifier": "no-style",
-  "description": "Caption text comes here.",
-  "h2_text": "Page wide title"
+  {% include '@figure' with {
+  "figure_image_src": "https://via.placeholder.com/1600x350",
+  "figure_image_alt_text": "Page wide image",
+  "modifier": 'page-wide-image',
+  "figure_figcaption": "Caption text comes here."
```
Comment (meirege) [inline styleguide/components/71-pages/general/info-page/info-page.twig]: Indentation.

File: styleguide/components/41-organisms/image-gallery/image-gallery.twig [hunk 8]
```diff
@@ -1,25 +1,13 @@
...
+<a href="{{ src }}"
+  {% if caption %}
+    aria-label="{{ caption }}"
+  {% endif %}
+   aria-describedby="image-gallery__open-gallery"
+   class="gallery-link">
+  {% include '@figure' with {
+  "figure_figcaption": caption,
+  "figure_image_src": src,
+  "figure_image_alt_text": alt
```
Comment (meirege) [inline styleguide/components/41-organisms/image-gallery/image-gallery.twig]: Indentation.

**Review by meirege — CHANGES_REQUESTED** [review]

File: styleguide/components/21-atoms/buttons/button-primary/button-primary.config.json (hunk 1 above)
Comment (delrueba) [reply styleguide/components/21-atoms/buttons/button-primary/button-primary.config.json:8]: No, it should.
Primary buttons are... primary. Should be first in the list. 
The others don't have an implicit order, as long as they come after primary and secondary.

File: styleguide/components/21-atoms/buttons/button-secondary/button-secondary.config.json (hunk 2 above)
Comment (delrueba) [reply styleguide/components/21-atoms/buttons/button-secondary/button-secondary.config.json:8]: see above

File: styleguide/components/21-atoms/headings/heading-6/_heading-6.scss (hunk 3 above)
Comment (delrueba) [reply styleguide/components/21-atoms/headings/heading-6/_heading-6.scss:8]: Neither are h7's...
Readme currently states '* Currently, only Headings 1 to 3 are used.'
Suggest leaving this for the documentation update (in collaboration with UX).

File: styleguide/components/21-atoms/inputs/inputs.config.json (hunk 4 above)
Comment (delrueba) [reply styleguide/components/21-atoms/inputs/inputs.config.json:3]: Yes, I don't want to cause confusion by showing input fields without labels.
The same reason we hide 'figcaption'.. they just cannot be used by themselves.

Now if you search 'input' the first thing you see is something which should not be used...

Do agree we should carefully document the molecules to explain which parts are optional and which not. But I'd leave that for the documentation ticket.

File: styleguide/components/41-organisms/header/header--with-menu-and-banner.twig (hunk 6 above)
Comment (delrueba) [reply styleguide/components/41-organisms/header/header--with-menu-and-banner.twig:61]: done

File: styleguide/components/21-atoms/buttons/button-primary/button-primary.config.json (hunk 1 above)
Comment (meirege) [reply styleguide/components/21-atoms/buttons/button-primary/button-primary.config.json:8]: Okay sounds fair.

File: styleguide/components/21-atoms/buttons/button-secondary/button-secondary.config.json (hunk 2 above)
Comment (meirege) [reply styleguide/components/21-atoms/buttons/button-secondary/button-secondary.config.json:8]: Okay sounds fair.

File: styleguide/components/21-atoms/headings/heading-6/_heading-6.scss (hunk 3 above)
Comment (meirege) [reply styleguide/components/21-atoms/headings/heading-6/_heading-6.scss:8]: Ok.

File: styleguide/components/21-atoms/inputs/inputs.config.json (hunk 4 above)
Comment (meirege) [reply styleguide/components/21-atoms/inputs/inputs.config.json:3]: Ok agreed, this seems like a good approach.

**Review by meirege — APPROVED** [review]

**Comment by meirege** [comment]
@delrueba Just the codeclimate issue.

**Pull request merged**